### PR TITLE
project: Fix and check pyupgrade issues

### PR DESCRIPTION
**Pull request by pdgendt**
Depending on the minimum supported python version (3.9 in our case), `pyupgrade` will report deprecated python features.
Extend `ruff` with `pyupgrade` checks.

Depends on #762

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -200,9 +200,9 @@ def repos_tmpdir(tmpdir, _session_repos):
         path: zephyr
 
     '''
-    kconfiglib, tagged_repo, net_tools, zephyr = [
+    kconfiglib, tagged_repo, net_tools, zephyr = (
         os.path.join(_session_repos, x) for x in
-        ['Kconfiglib', 'tagged_repo', 'net-tools', 'zephyr']]
+        ['Kconfiglib', 'tagged_repo', 'net-tools', 'zephyr'])
```
Comment (marc-hb) [inline tests/conftest.py:205]: I'm curious what was the warning here...

**Review by marc-hb — COMMENTED** [review]
The cleanup commit is a bit too big for my taste hence a bit painful to check... could you split it in say 2 or 3? Maybe one commit with all the type changes and another commit for the rest...

File: tests/conftest.py (hunk 1 above)
Comment (pdgendt) [reply tests/conftest.py]: It was actually a [deprecated rule](https://docs.astral.sh/ruff/rules/unpacked-list-comprehension/) itself, which was still "automagically" fixed with `ruff check --fix`.

I've ignored this rule.

**Review by carlescufi — APPROVED** [review]
LGTM, thanks!

**Review by marc-hb — APPROVED** [review]

**Pull request merged**